components/suggest/src/suggest.udl

```diff
@@ -69,6 +69,14 @@ interface Suggestion {
     );
 };
 
+enum SuggestRecord {
```

Comment:
Instead of surfacing `SuggestRecord` and "record types" to consumers, how would you feel about letting them specify providers instead? And then, in our component, we can do the mapping of provider to records.

So `SuggestionProvider.Amp` and `SuggestionProvider.Wikipedia` would fetch both `SuggestRecord.AmpWikipedia` and `SuggestRecord.Icon` records, `SuggestionProvider.Pocket` would fetch `SuggestRecord.Pocket` records, and so on.

That way, consumers won't need to know about our Remote Settings storage scheme, and we can switch it around—split up AMP and Wikipedia, remove icons and host them on a different server, remove attachments altogether—without them knowing.

Reply:
> So SuggestionProvider.Amp and SuggestionProvider.Wikipedia would fetch both SuggestRecord.AmpWikipedia and SuggestRecord.Icon records,

Ah cool this would work. I wasn't sure if what we would do with Icons if I used providers, which is why I opted for more control of records. If we know what the mapping looks like, I can do the Provider -> Record mapping.